build/ci/check_demo_pr_checks.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -xe
+set +o pipefail
+
+export passed_k8s_checks=true
+export passed_openshift_checks=true
+
+wait_for_checks_to_complete(){
+  community_operators_branch=$1
+  is_summary_test_passed=false
+  export repo_pr=`gh pr list --repo $github_csiblock_community_operators_repository | grep $community_operators_branch`
+  if [[ "$repo_pr" == *"$community_operators_branch"* ]]; then
+    sleep 5
+    while [ "$(gh pr checks $community_operators_branch --repo $github_csiblock_community_operators_repository | grep -i pending | wc -l)" -gt 0 ]; do
+      sleep 15
+    done
+    summary_counter=0
+    export search_summary_test=`gh pr checks $community_operators_branch --repo $github_csiblock_community_operators_repository | grep -i summary`
+    while [ "$search_summary_test" != *"pending"* ] && [ $summary_counter -lt 20 ] && [ $is_summary_test_passed == "false" ]; do
+      sleep 1
+      export search_summary_test=`gh pr checks $community_operators_branch --repo $github_csiblock_community_operators_repository | grep -i summary`
+      ((summary_counter=summary_counter+1))
+      if [[ "$search_summary_test" == *"pass"* ]]; then
+        is_summary_test_passed=true
+      fi
+    done
+    if [[ $is_summary_test_passed == "false" ]]; then
+      while [ "$(gh pr checks $community_operators_branch --repo $github_csiblock_community_operators_repository | grep -i pending | wc -l)" -gt 0 ]; do
+      sleep 5
```

Comment:
how can this be, after the [same `while`](https://github.com/IBM/ibm-block-csi-operator/blob/f58e6f493eeba993456e6786e97eb3cbb6c08c78/build/ci/check_demo_pr_checks.sh#L13) already finished?

Reply:
there are 3 `while`s - I'm not referring to the second one.
why do we need this 3rd `while`?

Reply:
I know, I meant to the third while

Reply:
is it a summary test or a test summary?
since the `while` conditions are identical - what would happen if the 1st `while` will start after the summary already finished?